### ISSUE TITLE
future enhancements?

### ISSUE THREAD

**Issue by lyolyalya** [aaronsace/MultiPoolMiner#51]
Hey...i was thinking...
do you know about NVIDIA-SMI utility?it shows power usage, temperatures, tdp and other cool things...
sooo,you know?how about to add monitoring of powerusage and temperatures?=)maybe hourly, daily,weekly, monthly predicitions based on current powerusage?if we can set power cost then we can predict our looses for electricity...
and now most intersting thing i thought...
nvidia-smi can set tpd...sooo...how abot benchamarking not just hashrate but benchmarking mh/w?
example.we can start test from 60% with 5 or 10 % increment to 100% and with powerusage can choose the best ratio mh/w? isn't it the dream of every miner?=)
ahhh...again those dreams...=)
how do you think?

**Comment by aaronsace** [comment]
That certainly sounds like what we need! 
The figure doesn't need to be 100% accurate but relative to the other benchmarks.

**Comment by lyolyalya** [comment]
well.hope to see this features soon=)if you neet to test some - be my guest=)

**Comment by snakethejake81** [comment]
It would be nice to see a GPU status table on the main refresh, showing temperature and maybe gpu/mem clocks and power usage? Per GPU.

This exports all current stats for NVIDIA cards installed in an XML format file:
nvidia-smi.exe --query --filename=C:\crypto\nvidia-smi.xml --xml-format

Trying to find an AMD equivalent still.

**Comment by aaronsace** [comment]
I could add all that info, was just going to read wattage for loss calculation.

**Comment by lyolyalya** [comment]
i think temperatures, power consumptions, and mh/W will be fine for now...maybe in future some predictions for monthly power cost...

**Comment by casperoid182** [comment]
Hey. Why release not have cryptonight algoritm for cpu? Now, i think, only monero give stable profit on cpu. So, if claymore-cpu-cryptonight miner will be added, it will be nice ))

**Comment by aaronsace** [comment]
There is one included already (not Claymore)

I'm adding a better one but it's a pain to add because the username has to be entered into a file?! It'd take 5 mins for the developer just to add it and would save all the hassle.

**Comment by casperoid182** [comment]
Yes, it re-benchmark that one miner for cryptonight. But when bechmark end, in stat file speed again 0:
Cpuminer_Cryptonight_HashRate:
{
    "Live":  0,
    "Minute":  0,
    "Minute_Fluctuation":  0.5,
    "Minute_5":  0,
    "Minute_5_Fluctuation":  0.5,
    "Minute_10":  0,
    "Minute_10_Fluctuation":  0.5,
    "Hour":  0,
    "Hour_Fluctuation":  0.5,
    "Day":  0,
    "Day_Fluctuation":  0.5,
    "Week":  0,
    "Week_Fluctuation":  0.5,
    "Updated":  "\/Date(1487778020828)\/"
}
cpu AMD fx-6300, in claymore miner 260h/s for cryptonight.

**Comment by aaronsace** [comment]
Please post as a separate issue. 

Delete the stat file. 
Run MultiPoolMiner. 
Send the log and a screenshot.

**Comment by casperoid182** [comment]
OK
[2017-02-22_10-03-43.txt](https://github.com/aaronsace/MultiPoolMiner/files/794684/2017-02-22_10-03-43.txt)
![testing](https://cloud.githubusercontent.com/assets/25950644/23230336/3922ce3e-f94c-11e6-9daa-d40558ca8afe.jpg)
![aftertest](https://cloud.githubusercontent.com/assets/25950644/23230342/3f3d781e-f94c-11e6-9d16-8dc1162c8586.jpg)
